src/common/util/util.ts

```diff
@@ -219,6 +219,41 @@ export function mapLazy<T, R>(xs: Iterable<T>, f: (x: T) => R): Iterable<R> {
   };
 }
 
+const Orders = {
+  forward: true,
+  backward: true,
+  shiftByHalf: true,
+};
+export type Order = keyof typeof Orders;
```

Comment:
nit: order seems a bit generic. Also is it really needed since reorder could type `order` as 
"forward" | "backward" | "shiftByHalf"? (if there's a `export ReorderOrderType = [...] as const`?)

Reply:
I'm not sure what you're suggesting here. Get rid of `reorder`? rename `Order` to `ReorderOrderType`? make reorder just take 3 strings?

Reply:
I renamed it `ReorderOrderType`. Happy to change it

Reply:
Thanks, that works, I was mostly worried about the overly generic name confusing people.